src/lib.rs

```diff
@@ -18,27 +18,25 @@ mod mime_types;
 /// Guess the MIME type of `path` by its extension (as defined by `Path::extension()`).
 ///
 /// If `path` has no extension, or its extension has no known MIME type mapping,
-/// then the MIME type is assumed to be `application/octet-stream`.
+/// then None is returned.
 ///
 /// ##Note
 /// **Guess** is the operative word here, as there are no guarantees that the contents of the file
 /// that `path` points to match the MIME type associated with the path's extension.
 ///
 /// Take care when processing files with assumptions based on the return value of this function.
-pub fn guess_mime_type<P: AsRef<Path>>(path: P) -> Mime {
```

Comment:
This should probably be renamed to `mime_type_from_ext` or `guess_mime_type_opt` and then `guess_mime_type` should remain as returning `Mime` for convenience:

```rust
pub fn guess_mime_type<P: AsRef<Path>>(path: P) -> Mime { 
    mime_type_from_ext(path).unwrap_or_else(octet_stream)
}
```